### ISSUE TITLE
[Web 3.24.0] canvaskit loading not honoring --base-href when using --no-web-resources-cdn

### ISSUE THREAD

**Issue by akaiser** [flutter/flutter#153340]
### Steps to reproduce

1. flutter create --platforms=web example
2. cd example/
3. flutter build web --no-web-resources-cdn --base-href /web/
4. python3 -m http.server 9000 -d build/
5. open: http://localhost:9000/web/
6. observe 404s on: http://localhost:9000/canvaskit/chromium/canvaskit.js and .wasm

Possibly related to: https://github.com/flutter/flutter/pull/150806

### Expected results

http://localhost:9000/web/canvaskit/chromium/canvaskit.js
and
http://localhost:9000/web/canvaskit/chromium/canvaskit.wasm

are called with specified /web/ prefix

### Actual results

http://localhost:9000/canvaskit/chromium/canvaskit.js
and
http://localhost:9000/canvaskit/chromium/canvaskit.wasm

resulting in 404s because not containing the specified /web/ prefix

### Code sample

See steps to reproduce

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

3.24.0 vs. 3.22.3 comparison

<img width="1535" alt="Screenshot 2024-08-13 at 09 40 45" src="https://github.com/user-attachments/assets/bfb33ecc-04cf-4017-adf1-cb0e93faa1b9">

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.24.0, on macOS 14.5 23F79 darwin-arm64, locale en-US)
    • Flutter version 3.24.0 on channel stable at /Users/albert.kaiser/Dev/_rt/flutter_3.24
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 80c2e84975 (2 weeks ago), 2024-07-30 23:06:49 +0700
    • Engine revision b8800d88be
    • Dart version 3.5.0
    • DevTools version 2.37.2

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/albert.kaiser/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/albert.kaiser/Library/Android/sdk
    • Java binary at: /Library/Java/JavaVirtualMachines/jdk-17.jdk/Contents/Home/bin/java
    • Java version Java(TM) SE Runtime Environment (build 17.0.8+9-LTS-211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15E204a
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/to/macos-android-setup for detailed instructions).

[✓] IntelliJ IDEA Ultimate Edition (version 2024.1.5)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 81.0.2
    • Dart plugin version 241.18968.26

[✓] VS Code (version 1.89.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.88.0

[✓] Connected device (3 available)
    • macOS (desktop)                 • macos                 • darwin-arm64   • macOS 14.5 23F79 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad • darwin         • macOS 14.5 23F79 darwin-arm64
    • Chrome (web)                    • chrome                • web-javascript • Google Chrome 127.0.6533.100

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

**Comment by huycozy** [comment]
Thanks for the report! I also see the same issue on Flutter stable 3.24.0. It seems to be fixed on master channel `Channel master, 3.24.0-1.0.pre.559` as I don't see it anymore but I see the below error instead. Can you also try and confirm this?

<details open>
<summary>Output log</summary>

```console
Uncaught (in promise) null
Promise.then
n @ flutter_bootstrap.js:1
didCreateEngineInitializer @ flutter_bootstrap.js:1
(anonymous) @ main.dart.js:6254
(anonymous) @ main.dart.js:4226
$2 @ main.dart.js:25839
O @ main.dart.js:4212
a_H @ main.dart.js:6256
(anonymous) @ main.dart.js:17578
(anonymous) @ main.dart.js:4226
$2 @ main.dart.js:25839
O @ main.dart.js:4212
a0b @ main.dart.js:17580
(anonymous) @ main.dart.js:56096
(anonymous) @ main.dart.js:56092
dartProgram @ main.dart.js:56094
(anonymous) @ main.dart.js:56096
```
</details>

**Comment by akaiser** [comment]
Hey @huycozy! Confirming that master `3.24.0-1.0.pre.563` loads expected! But I do not confirm seeing any errors in the console! All clear.

**Comment by huycozy** [comment]
Thank you for confirming this. Looks like it's a cache issue on HTTP server. When I change the port, the web is loaded well without any issues. 

Closing this as fixed on master channel.

**Comment by akaiser** [comment]
@huycozy thx! I assume correctly that the fix also will be part of `3.24.1` => https://github.com/flutter/flutter/issues/152953 ?

**Comment by huycozy** [comment]
It could be the same cause with those issues. You can ask the same there for sure.
